### DISCUSSION TITLE
[V5] Block other access by the same user in the same application

### DISCUSSION THREAD

**Discussion by roliveira3891** [adonisjs/core#2294] [Help] [answered]
Hello everybody. Help please. I have an application made in AdonisJS with Auth Session. However there are some users who share their access. How to block for a user not to access more than one session? if he logs in on that ip, logout on the other. Thanks.

**Comment by McSneaky** [comment, marked as answer]
There are multiple ways to do it
You can check for IP
Or you can make it so that only one session is allowed per user (save session state somewhere, like DB or Redis) and when new session is started then invalidate the previous one eventually logging the previous one out
Or use the same, to not allow new signin. But that might be problematic when user is legit changing computer / location
I'd go with signing out all other sessions and then creating new one at new login

**Reply by roliveira3891** [reply]
I did a middleware checking the session.sessionId in the database. It was show, Thanks.
